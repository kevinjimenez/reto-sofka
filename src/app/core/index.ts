export { spinnerInterceptor } from './interceptors/spinner.interceptor';
export { productsResolver } from './resolvers/products.resolver';
export { ProductsService } from './services/products.service';
export { SpinnerService } from './services/spinner.service';
