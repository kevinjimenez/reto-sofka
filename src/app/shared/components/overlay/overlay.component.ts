import { Component, model } from '@angular/core';

@Component({
	selector: 'app-overlay',
	standalone: true,
	imports: [],
	templateUrl: './overlay.component.html',
	styleUrl: './overlay.component.css'
})
export class OverlayComponent {
	public isVisible = model<boolean>(false);
}
