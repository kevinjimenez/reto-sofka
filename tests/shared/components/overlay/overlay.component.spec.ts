import { ComponentFixture, TestBed } from '@angular/core/testing';
import { OverlayComponent } from '../../../../src/app/shared/components/overlay/overlay.component';

describe('OverlayComponent', () => {
	let component: OverlayComponent;
	let fixture: ComponentFixture<OverlayComponent>;
	let compiled: HTMLElement;

	beforeEach(async () => {
		await TestBed.configureTestingModule({
			imports: [OverlayComponent]
		}).compileComponents();
	});

	beforeEach(() => {
		fixture = TestBed.createComponent(OverlayComponent);
		component = fixture.componentInstance;
		fixture.detectChanges();
		compiled = fixture.nativeElement;
	});

	it('should create', () => {
		expect(component).toBeTruthy();
	});

	it('should match snapshot', () => {
		expect(compiled).toMatchSnapshot();
	});

	it('should render the content', () => {
		component.isVisible.set(true);
		fixture.detectChanges();

		const contend = compiled.querySelector('.overlay-content');
		expect(contend).toBeTruthy();
	});
});
