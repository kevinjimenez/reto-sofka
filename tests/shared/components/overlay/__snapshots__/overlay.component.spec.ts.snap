// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OverlayComponent should match snapshot 1`] = `
<div
  id="root1"
>
  
</div>
`;
